### PR TITLE
refactor: use seed for sample for consistency

### PR DESCRIPTION
**Pull request by JGSweets**
This PR:
 * refactors to use seed when sampling for consistency
 * Fixes structuredprofiler tests to include for corr, null_mat, chi2
 * fixes decorators to make things faster

### CODE REVIEW

File: dataprofiler/profilers/profile_builder.py [hunk 1]
```diff
@@ -645,11 +646,22 @@ def clean_data_and_get_base_stats(
...
+                warnings.warn("Seed should be an integer", RuntimeWarning)
+
         base_stats = {
             "sample_size": total_sample_size,
             "null_count": total_na,
             "null_types": na_columns,
-            "sample": random.sample(list(df_series.values), min(len(df_series), 5)),
+            "sample": rng.choice(
+                list(df_series.values), (min(len(df_series), 5),), replace=False
+            ).tolist(),
```
Comment (JGSweets) [inline dataprofiler/profilers/profile_builder.py:664]: the big fix is here for making sampling consistent

File: dataprofiler/tests/profilers/test_profile_builder.py [hunk 2]
```diff
@@ -125,7 +124,7 @@ def test_list_data(self, *mocks):
 
         # validates the sample out maintains the same visual data format as the
         # input.
-        self.assertListEqual(["5", "1", "1", "3", "4"], profiler.profile[0].sample)
```
Comment (JGSweets) [inline dataprofiler/tests/profilers/test_profile_builder.py:128]: bc consistent now, changed order

File: dataprofiler/tests/profilers/test_profile_builder.py [hunk 3]
```diff
@@ -2273,7 +2272,17 @@ def test_json_decode(self, mock_utils_DataLabeler, mock_DataLabeler, *mocks):
         mock_DataLabeler.return_value = mock_labeler
 
         fake_profile_name = None
-        expected_profile = StructuredProfiler(fake_profile_name)
+        profile_options = dp.ProfilerOptions()
```
Comment (JGSweets) [inline dataprofiler/tests/profilers/test_profile_builder.py:2275]: make things faster here

File: dataprofiler/tests/profilers/test_profile_builder.py [hunk 4]
```diff
@@ -2309,7 +2318,17 @@ def test_json_decode_after_update(
             "pred": [],
             "conf": [[1, 1], [0, 0]],
         }
-        expected_profile = StructuredProfiler(fake_profile_name)
```
Comment (JGSweets) [inline dataprofiler/tests/profilers/test_profile_builder.py:2312]: make things faster here and enable missing stats

File: dataprofiler/tests/profilers/test_profile_builder.py [hunk 5]
```diff
@@ -2340,15 +2359,11 @@ def test_json_decode_after_update(
             ]
         )
 
-        deserialized.update_profile(df_structured)
+        with test_utils.mock_timeit():
```
Comment (JGSweets) [inline dataprofiler/tests/profilers/test_profile_builder.py:2362]: mock to ensure times the same

File: dataprofiler/tests/profilers/test_profile_builder.py [hunk 6]
```diff
@@ -2340,15 +2359,11 @@ def test_json_decode_after_update(
...
+            expected_profile.update_profile(df_structured)
 
-        assert deserialized.total_samples == 5
-        assert deserialized._max_col_samples_used == 5
-        assert deserialized._min_col_samples_used == 5
-        assert deserialized.row_has_null_count == 2
-        assert deserialized.row_is_null_count == 1
-        assert deserialized._get_unique_row_ratio() == 0.80
-        assert deserialized.file_type == "<class 'pandas.core.frame.DataFrame'>"
+        test_utils.assert_profiles_equal(deserialized, expected_profile)
```
Comment (JGSweets) [inline dataprofiler/tests/profilers/test_profile_builder.py:2366]: now just validate that both update separately get same values

File: dataprofiler/tests/profilers/test_profile_builder.py [hunk 7]
```diff
@@ -2523,7 +2528,7 @@ def test_clean_data_and_get_base_stats(self, *mocks):
         self.assertTrue(np.issubdtype(np.object_, df_series.dtype))
         self.assertDictEqual(
             {
-                "sample": ["4.0", "6.0", "3.0"],
```
Comment (JGSweets) [inline dataprofiler/tests/profilers/test_profile_builder.py:2526]: bc consistent now, changed order

File: dataprofiler/tests/profilers/test_profile_builder.py [hunk 8]
```diff
@@ -2540,7 +2545,7 @@ def test_clean_data_and_get_base_stats(self, *mocks):
         )
         self.assertDictEqual(
             {
-                "sample": ["nan", "6.0", "4.0", "nan"],
```
Comment (JGSweets) [inline dataprofiler/tests/profilers/test_profile_builder.py:2543]: bc consistent now, changed order

File: dataprofiler/tests/profilers/test_profile_builder.py [hunk 9]
```diff
@@ -2557,7 +2562,7 @@ def test_clean_data_and_get_base_stats(self, *mocks):
         )
         self.assertDictEqual(
             {
-                "sample": ["3.0", "4.0", "6.0", "nan", "1.0"],
```
Comment (JGSweets) [inline dataprofiler/tests/profilers/test_profile_builder.py:2560]: bc consistent now, changed order

File: dataprofiler/tests/profilers/test_profile_builder.py [hunk 10]
```diff
@@ -2696,11 +2699,9 @@ def test_index_overlap_for_update_profile(self, *mocks):
         self.assertDictEqual(profile.null_types_index, {"nan": {1, 4, 6, 9}})
 
     @mock.patch(
-        "dataprofiler.profilers.profile_builder." "ColumnPrimitiveTypeProfileCompiler"
+        "dataprofiler.profilers.column_profile_compilers.BaseCompiler.update_profile"
```
Comment (JGSweets) [inline dataprofiler/tests/profilers/test_profile_builder.py:2702]: fixed and now faster

File: dataprofiler/tests/profilers/test_profile_builder.py [hunk 11]
```diff
@@ -2721,11 +2722,9 @@ def test_index_overlap_for_merge(self, *mocks):
         self.assertDictEqual(profile2.null_types_index, {"nan": {1, 4}})
 
     @mock.patch(
-        "dataprofiler.profilers.profile_builder." "ColumnPrimitiveTypeProfileCompiler"
```
Comment (JGSweets) [inline dataprofiler/tests/profilers/test_profile_builder.py:2724]: fixed and now faster

File: dataprofiler/tests/profilers/test_profile_builder.py [hunk 12]
```diff
@@ -2748,10 +2747,13 @@ def test_min_max_id_properly_update(self, *mocks):
         self.assertEqual(0, profile._min_id)
         self.assertEqual(6, profile._max_id)
 
-    @mock.patch("dataprofiler.profilers.data_labeler_column_profile.DataLabeler")
     @mock.patch(
         "dataprofiler.profilers.data_labeler_column_profile.DataLabelerColumn.update"
     )
+    @mock.patch(
```
Comment (JGSweets) [inline dataprofiler/tests/profilers/test_profile_builder.py:2753]: fixed and now faster

File: dataprofiler/tests/profilers/test_profile_builder.py [hunk 13]
```diff
@@ -2884,7 +2886,7 @@ def test_json_encode_after_update(self, mock_DataLabeler, *mocks):
                 "_sampling_ratio": 0.2,
                 "_min_true_samples": 0,
                 "sample_size": 4,
-                "sample": ["1", "2", "-2"],
```
Comment (JGSweets) [inline dataprofiler/tests/profilers/test_profile_builder.py:2887]: bc consistent now, changed order

**Review by micdavis — APPROVED** [review]

File: dataprofiler/tests/profilers/test_profile_builder.py (hunk 6 above)
Comment (taylorfturner) [reply dataprofiler/tests/profilers/test_profile_builder.py:2497]: very nice -- I like

**Review by taylorfturner — APPROVED** [review]

**Pull request merged**